storage/src/backend/connection.rs

```diff
@@ -345,10 +354,37 @@ impl Connection {
                     Some(ReqBody::Buf(buf)) => Some(ReqBody::Buf(buf.clone())),
                     _ => None,
                 };
+
```

Comment:
It would be great to abstract this as a helper function.

Reply:
Good point, will add such a function ~

Reply:
added